rackcity/models/rack.py

```diff
@@ -49,3 +49,16 @@ def save(self, *args, **kwargs):
             raise valid_error
         else:
             super(Rack, self).save(*args, **kwargs)
+            self.add_pdu_ports()
+
+    def add_pdu_ports(self):
+        from rackcity.models import PDUPort
+        if len(PDUPort.objects.filter(rack=self.id)) == 0:
+            for left_right in ["L", "R"]:
+                for port_number in range(1, 25):  # 1 through 24 inclusive
+                    pdu_port = PDUPort(
+                        rack=self,
+                        left_right=left_right,
+                        port_number=port_number,
+                    )
+                    pdu_port.save()
```

Comment:
I think we need similar code on rack.delete() so that the rack's associated rows in the pduport table are also deleted?

Reply:
NICE CATCH I almost forgot this and then was proud of myself for realizing HOWEVER I tested and when you delete the rack all of the pdu ports get deleted as well which is the behavior we want.  Nice thing that we get for free from django/postgres given our new database strats

Reply:
that's crazy. how beautiful